src/cache_cf.cc

```diff
@@ -3879,88 +3863,72 @@ dump_generic_port(StoreEntry * e, const char *n, const AnyP::PortCfgPointer &s)
 
         if (s->defaultsite)
-            storeAppendPrintf(e, " defaultsite=%s", s->defaultsite);
+            out <<  "defaultsite=" << s->defaultsite << ' ';
 
         // TODO: compare against prefix of 'n' instead of assuming http_port
         if (s->transport.protocol != AnyP::PROTO_HTTP)
-            storeAppendPrintf(e, " protocol=%s", AnyP::ProtocolType_str[s->transport.protocol]);
+            out << "protocol=" << AnyP::ProtocolType_str[s->transport.protocol];
 
         if (s->allow_direct)
-            storeAppendPrintf(e, " allow-direct");
+            out << "allow-direct ";
 
         if (s->ignore_cc)
-            storeAppendPrintf(e, " ignore-cc");
+            out << "ignore-cc ";
 
     }
 
     // Generic independent options
 
     if (s->name)
-        storeAppendPrintf(e, " name=%s", s->name);
+        out << "name=" << s->name << ' ';
 
 #if USE_HTTP_VIOLATIONS
     if (!rawFlags.accelSurrogate && s->ignore_cc)
-        storeAppendPrintf(e, " ignore-cc");
+        out << "ignore-cc ";
 #endif
 
-    if (s->connection_auth_disabled)
-        storeAppendPrintf(e, " connection-auth=off");
-    else
-        storeAppendPrintf(e, " connection-auth=on");
-
-    if (s->disable_pmtu_discovery != DISABLE_PMTU_OFF) {
-        const char *pmtu;
+    out << "connection-auth=" << (s->connection_auth_disabled ? "off" : "on") << ' ';
 
-        if (s->disable_pmtu_discovery == DISABLE_PMTU_ALWAYS)
-            pmtu = "always";
-        else
-            pmtu = "transparent";
-
-        storeAppendPrintf(e, " disable-pmtu-discovery=%s", pmtu);
-    }
+    if (s->disable_pmtu_discovery != DISABLE_PMTU_OFF)
+        out << "disable-pmtu-discovery=" << (s->disable_pmtu_discovery == DISABLE_PMTU_ALWAYS ? "always" : "transparent")  << ' ';
 
     if (s->s.isAnyAddr() && !s->s.isIPv6())
-        storeAppendPrintf(e, " ipv4");
+        out << "ipv4 ";
 
     if (s->tcp_keepalive.enabled) {
-        if (s->tcp_keepalive.idle || s->tcp_keepalive.interval || s->tcp_keepalive.timeout) {
-            storeAppendPrintf(e, " tcpkeepalive=%d,%d,%d", s->tcp_keepalive.idle, s->tcp_keepalive.interval, s->tcp_keepalive.timeout);
-        } else {
-            storeAppendPrintf(e, " tcpkeepalive");
-        }
+        out << "tcpkeepalive";
+        if (s->tcp_keepalive.idle || s->tcp_keepalive.interval || s->tcp_keepalive.timeout)
+            out << "=" << s->tcp_keepalive.idle << ',' <<  s->tcp_keepalive.interval << ',' << s->tcp_keepalive.timeout;
+        out << ' ';
     }
 
-#if USE_OPENSSL
-    if (rawFlags.tunnelSslBumping)
-        storeAppendPrintf(e, " ssl-bump");
-#endif
+    auto outBuf = out.buf();
+    // Remove the trailing whitespace because ServerOptions::dumpCfg() below does not expect it.
+    // TODO: refactor to avoid this hack.
+    static const SBuf whiteSpace(" ");
+    outBuf.trim(whiteSpace, false, true);
+
+    e->append(outBuf.rawContent(), outBuf.length());
 
     s->secure.dumpCfg(e, "tls-");
 }
```

Comment:
If possible, please avoid most of dump_generic_port() changes by using a PackableStream to dump flags. For an example, see how SBufStatsAction::dump() combines modern ostream-based printing and legacy StoreEntry-based dumping. There are probably other, better examples in the code. Look for PackableStream.